### PR TITLE
Add support for 0x04 hex colors

### PR DESCRIPTION
**Pull request by xPaw**
Ref: https://modern.ircdocs.horse/formatting.html#hex-color

At least IRCCloud and AdiIRC support it.

cc @Bonuspunkt

### CODE REVIEW

File: client/js/libs/handlebars/ircmessageparser/parseStyle.js [hunk 1]
```diff
@@ -12,6 +13,9 @@ const UNDERLINE = "\x1f";
 // integers, `XX` is the text color and `YY` is an optional background color.
 const colorRx = /^(\d{1,2})(?:,(\d{1,2}))?/;
 
+// 6-char Hex color code matcher
+const hexColorRx = /^([a-fA-F0-6]{6})?/;
```
Comment (Bonuspunkt) [inline client/js/libs/handlebars/ircmessageparser/parseStyle.js]: pattern seems wrong, should be `/^([0-9a-f]{6})?/i` ?

File: test/client/js/libs/handlebars/ircmessageparser/parseStyle.js [hunk 2]
```diff
@@ -104,12 +109,70 @@ describe("parseStyle", () => {
 		expect(actual).to.deep.equal(expected);
 	});
 
+	it("should parse hex colors", () => {
+		const input = "test \x04FFFFFFnice \x02\x04RES006 \x0303,04\x04ff00FFcolored";
```
Comment (Bonuspunkt) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: is hex formatting foreground only?
also more color variation would be nice (see issue with pattern)

File: client/js/libs/handlebars/ircmessageparser/parseStyle.js (hunk 1 above)
Comment (xPaw) [inline client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Yeah that `0-6` is wrong.

File: test/client/js/libs/handlebars/ircmessageparser/parseStyle.js (hunk 2 above)
Comment (xPaw) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Yes, only foreground.

Comment (DanielOaks) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Hmm, AdiIRC seems to allow foreground and background. I should make it more clear on the doc that it allows setting both foreground and background colours with hex in this way

Comment (astorije) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: @xPaw, do you want/need to update your PR to account for foreground/background as per @DanielOaks' comment or not?

Comment (xPaw) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: This currently does not support background colors. Considering @DanielOaks didn't know that too, I wonder how many clients implement it, compared to just the foreground color.

Comment (astorije) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Right, it doesn't. But don't we want to be the first IRC client leading the way forward? :D (Ok ok it's a minor lead 😅)

Comment (xPaw) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Considering there is no formal definition, we want to know how it's already implemented in other clients.

Comment (astorije) [inline test/client/js/libs/handlebars/ircmessageparser/parseStyle.js]: Fair enough?

**Comment by xPaw** [comment]
Added support for background colors.

**Pull request merged**